src/internal_events/amqp.rs

```diff
@@ -5,24 +5,56 @@ pub mod source {
     use vector_core::internal_event::InternalEvent;
 
     #[derive(Debug)]
-    pub struct AMQPEventReceived {
+    pub struct AmqpBytesReceived {
         pub byte_size: usize,
+        pub protocol: &'static str,
     }
 
-    impl InternalEvent for AMQPEventReceived {
+    impl InternalEvent for AmqpBytesReceived {
         fn emit(self) {
-            trace!(message = "Received one event.", internal_log_rate_secs = 10);
-            counter!("processed_events_total", 1);
-            counter!("processed_bytes_total", self.byte_size as u64);
+            trace!(
+                message = "Bytes received.",
+                byte_size = %self.byte_size,
+                protocol = %self.protocol,
+            );
+            counter!(
+                "component_received_bytes_total",
+                self.byte_size as u64,
+                "protocol" => self.protocol,
+            );
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct AmqpEventsReceived {
+        pub byte_size: usize,
+        pub count: usize,
+    }
+
+    impl InternalEvent for AmqpEventsReceived {
+        fn emit(self) {
+            trace!(
+                message = "Events received.",
+                count = self.count,
+                byte_size = self.byte_size
+            );
+            counter!("component_received_events_total", 1);
+            counter!(
+                "component_received_event_bytes_total",
+                self.byte_size as u64
+            );
+
+            // deprecated
+            counter!("events_in_total", 1);
```

Comment:
Just curious what the guideline is here- since this is a new component. Shouldn't we not add deprecated metrics?

Reply:
It is no more.